conclave-host/src/main/kotlin/com/r3/conclave/host/internal/gramine/GramineEnclaveHandle.kt

```diff
@@ -150,20 +158,46 @@ class GramineEnclaveHandle(
         return false
     }
 
+    private fun unTarFile(tarGz: Path, outputDir: Path) {
+        tarGz.inputStream().use { fis ->
+            GZIPInputStream(fis).use { gis ->
+                TarArchiveInputStream(gis).use { tis ->
```

Comment:
Closing `TarArchiveInputStream` will close all the underlying streams as well. So you can simplify this to
```
TarArchiveInputStream(GZIPInputStream(tarGz.inputStream())).use { tar ->

}
```